app/assess/templates/assessor_tasklist.html

```diff
@@ -5,39 +5,44 @@
 {% from 'macros/criteria_element.html' import criteria_element -%}
 {% from "macros/banner_summary.html" import banner_summary %}
 {% from "macros/flag_application_button.html" import flag_application_button %}
+{% from "macros/mark_qa_complete_button.html" import mark_qa_complete_button %}
 {% from "macros/assessment_flag.html" import assessment_flagged %}
 {% from "macros/assessment_flag.html" import assessment_stopped %}
 {% from "macros/logout_partial.html" import logout_partial %}
+{% from "macros/qa_complete_flag.html" import qa_complete_flag %}
+
 
 {% block content %}
-    <div class="govuk-phase-banner flex-parent-element">
-        <p class="govuk-!-text-align-left flex-parent-element flexed-element-margins-collapse">
-            {{ govukBackLink({'href': url_for("assess_bp.landing"), 'text': 'Back to assessment dashboard'}) }}
-        </p>
-        {{ logout_partial(sso_logout_url) }}
-    </div>
 
-    {% if flag and current_user_role in ("ASSESSOR", "LEAD_ASSESSOR") and state.display_status == "FLAGGED" %}
+<div class="govuk-phase-banner flex-parent-element">
+    <p class="govuk-!-text-align-left flex-parent-element flexed-element-margins-collapse">
+        {{ govukBackLink({'href': url_for("assess_bp.landing"), 'text': 'Back to assessment dashboard'}) }}
+    </p>
+    {{ logout_partial(sso_logout_url) }}
+</div>
+
+    {% if flag and current_user_role in ("LEAD_ASSESSOR") and flag.is_qa_complete %}
+    {{ qa_complete_flag(flag, flag_user_info) }}
+    {% endif %}
+
+    {% if flag and current_user_role in ("ASSESSOR", "LEAD_ASSESSOR") and flag.flag_type.name == "FLAGGED" %}
         {{ assessment_flagged(flag, flag_user_info, application_id, current_user_role) }}
     {% elif flag and current_user_role in ("ASSESSOR", "LEAD_ASSESSOR") and state.display_status == "STOPPED" %}
         {{ assessment_stopped(flag, flag_user_info, application_id, current_user_role) }}
-    {% endif %}
-
-    {% if state.workflow_status=="COMPLETED" %}
-    <div class="govuk-grid-row">
-        <div class="govuk-grid-column-full">
-            <div class="assessment-complete  govuk-margin-bottom-8">
-                <h2 class="assessment-alert__heading govuk-heading-l govuk-!-margin-top-4">
-                    Assessment complete</h2>
-                <p class="govuk-body">You have marked this assessment as complete.</p>
-                <p class="govuk-body">
-                    It may now be selected for quality assurance (QA). New scores and rationales could be added if this
-                    happens.
-                </p>
+    {% elif state.workflow_status=="COMPLETED" and not flag.is_qa_complete %}
```

Comment:
This refactored code block seems to not show assessment complete if there is a flag, I think this is contrary to seeing flags and the completion message on the same page as mentioned here https://communities-govuk.slack.com/archives/C04L4H0218U/p1674645287629009. *We should be able to run a flag workflow on a completed application.

Reply:
done!